addons/mail/static/src/core/common/message_model.js

```diff
@@ -206,24 +207,11 @@ export class Message extends Record {
     }
 
     get dateSimpleWithDay() {
-        const userLocale = { locale: user.lang };
-        if (this.datetime.hasSame(DateTime.now(), "day")) {
-            return _t("Today at %(time)s", {
-                time: this.datetime.toLocaleString(DateTime.TIME_SIMPLE, userLocale),
-            });
-        }
-        if (this.datetime.hasSame(DateTime.now().minus({ day: 1 }), "day")) {
-            return _t("Yesterday at %(time)s", {
-                time: this.datetime.toLocaleString(DateTime.TIME_SIMPLE, userLocale),
-            });
-        }
-        if (this.datetime?.year === DateTime.now().year) {
-            return this.datetime.toLocaleString(
-                { ...DateTime.DATETIME_MED, year: undefined },
-                userLocale
-            );
-        }
-        return this.datetime.toLocaleString({ ...DateTime.DATETIME_MED }, userLocale);
+        return toDateAt(this.datetime);
+    }
+
+    get editedDateSimpleWithDay() {
+        return _t("Last edited %(date)s", { date: toDateAt(this.write_date) });
```

Comment:
Would put colon `:` in-between "Last edited" and the date, because `toDateAt` starts with capital letter, and also this splits the 2 part of string in a way that works in other languages.